tests_v2/unit/app/routes/datasets/test_query.py

```diff
@@ -268,12 +268,17 @@ async def test_query_dataset_raster_geostore_huge(
 
 @pytest.mark.asyncio()
 async def test_query_vector_asset_disallowed_1(
-    generic_vector_source_version, async_client: AsyncClient
+    generic_vector_source_version, apikey, async_client: AsyncClient
 ):
     dataset, version, _ = generic_vector_source_version
 
+    api_key, payload = apikey
+    origin = "https://" + payload["domains"][0]
+    headers = {"origin": origin, "x-api-key": api_key}
+
```

Comment:
Would be nice to make a single function get_header that replaces these three lines everywhere with `headers = get_header(apikey)`